### PR TITLE
Remove obsolete TODOs from xplat SslStream code

### PR DESCRIPTION
**Pull request by shrutigarg**
This includes removing TODOs from the places which are already investigated and fixed . 
There are few places where TODO was kept but after investigation no changes needed.
At few places updated the TODO with new issue id where we have scope to further investigate and fix.

### CODE REVIEW

**Comment by shrutigarg** [comment]
@vijaykota @stephentoub @CIPop @rajansingh10 @bartonjs @eerhardt

**Comment by eerhardt** [comment]
LGTM.

File: src/System.Net.Security/src/System/Net/CertificateValidationPal.Unix.cs [hunk 1]
```diff
@@ -246,7 +246,8 @@ private static int QueryContextRemoteCertificate(SafeDeleteContext securityConte
 
         private static int QueryContextIssuerList(SafeDeleteContext securityContext, out Object issuerList)
         {
-            // TODO (Issue #3362) To be implemented
+            // Code to get the Issuer List is in GetRequestCertificateAuthorities()
+            // Being a PAL function , this needs to be present but not used anywhere.
```
Comment (stephentoub) [inline src/System.Net.Security/src/System/Net/CertificateValidationPal.Unix.cs]: This is the only "QueryContextIssuerList" I see being mentioned anywhere in all of corefx.  Can this function just be deleted entirely?

File: src/System.Net.Security/src/System/Net/_SslSessionsCache.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ internal SslCredKey(byte[] thumbPrint, int allowedProtocols, bool serverMode, En
 
                 _HashCode ^= allowedProtocols;
                 _HashCode ^= (int)encryptionPolicy;
-                _HashCode ^= serverMode ? 5 : 7; //TODO (Issue #3362) used a prime number here as it's a XOR. Figure out appropriate value.
+                _HashCode ^= serverMode ? 5 : 7; // Used a prime number here as it's a XOR. The values should not conflict with EncryptionPolicy and other parts of the hash
```
Comment (CIPop) [inline src/System.Net.Security/src/System/Net/_SslSessionsCache.cs]: @bartonjs PTAL

**Comment by CIPop** [comment]
LGTM

File: src/System.Net.Security/src/System/Net/_SslSessionsCache.cs (hunk 2 above)
Comment (bartonjs) [inline src/System.Net.Security/src/System/Net/_SslSessionsCache.cs]: I'm not really certain why it's setting the 0 and 2 bits, and sometimes also the 1 bit.  `serverMode ? 0 : 2` is maybe better, and trying to twiddle some bits that aren't twiddled by encryptionPolicy or allowedProtocols is best (unless your hash table uses a power-of-two of buckets, in which case only the low bits matter).

Comment (shrutigarg) [inline src/System.Net.Security/src/System/Net/_SslSessionsCache.cs]: 0 and 2 we cannot anyways use as these numbers already represent EncryptionPolicy ..
If we want to maintain the unique bits altogether ,, we can use the number beyond the sslprotocols values
i.e:
using 0x10000 and 0x20000 ..

File: src/System.Net.Security/src/System/Net/CertificateValidationPal.Unix.cs (hunk 1 above)
Comment (shrutigarg) [inline src/System.Net.Security/src/System/Net/CertificateValidationPal.Unix.cs]: removed.

File: src/System.Net.Security/src/System/Net/_SslSessionsCache.cs (hunk 2 above)
Comment (shrutigarg) [inline src/System.Net.Security/src/System/Net/_SslSessionsCache.cs]: @bartonjs PTAL

File: src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs [hunk 3]
```diff
@@ -279,6 +279,6 @@ public override string ToString()
 
     internal abstract class SafeFreeContextBufferChannelBinding : ChannelBinding
     {
-        // TODO (Issue #3362) To be implemented
+        // TODO (Issue #3954) To be implemented
```
Comment (vijaykota) [inline src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs]: This will need a rebase

Comment (shrutigarg) [inline src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs]: will do . before merge

Comment (shrutigarg) [inline src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs]: done rebase

**Comment by stephentoub** [comment]
One comment, otherwise LGTM. Please squash before merging.

**Pull request merged**